src/mon/OSDMonitor.cc

```diff
@@ -4599,6 +4599,14 @@ int OSDMonitor::prepare_new_pool(string& name, uint64_t auid,
 {
   if (name.length() == 0)
     return -EINVAL;
+
+  for (map<int64_t,string>::iterator p = pending_inc.new_pool_names.begin();
+       p != pending_inc.new_pool_names.end();
+       ++p) {
+    if (p->second == name)
+      return 0;
+  }
+
```

Comment:
So if we receive duplicated request of pool-creation request, we can fast catch it、skip over unnecessary sanity check and make quick response.